### ISSUE TITLE
Error Invalid Semver Range for npm: alias protocol

### ISSUE THREAD

**Issue by danielpza** [JamieMason/syncpack#151]
## Description

Syncpack is throwing an error when encountering a npm: alias protocol in the package.json definition

```
TypeError: Invalid SemVer Range: npm:@mui/styled-engine-sc@^5.12.0
```

I'm using [@mui/material](https://mui.com/) with styled-components, this is a valid package.json dependency entry:

```json
		"@mui/material": "5.13.6",
		"@mui/styled-engine": "npm:@mui/styled-engine-sc@^5.12.0",
		"@mui/styled-engine-sc": "5.12.0",
```

See https://mui.com/material-ui/guides/styled-engine/

## Suggested Solution

Syncpack should recognize the npm: alias protocol https://yarnpkg.com/features/protocols

**Comment by JamieMason** [comment]
Looking at https://docs.npmjs.com/cli/v9/using-npm/package-spec, it lists a lot of variations of package specs supported by npm, which we can use as a checklist for what to support. 

Importantly, it also lists https://www.npmjs.com/package/npm-package-arg which is new to me. It looks like it could be the missing piece for a lot of open issues, as it handles the awkward protocols (npm:, workspace:, link: etc) which are out of scope for https://github.com/npm/node-semver, currently used by syncpack.

**Comment by JamieMason** [comment]
Released in [10.9.3](https://github.com/JamieMason/syncpack/releases/tag/10.9.3), more work may be needed but let me know how this goes.

**Comment by JamieMason** [comment]
Update: [12.0.0-alpha.0](https://github.com/JamieMason/syncpack/releases/tag/12.0.0-alpha.0) adds better support for non-semver version formats.
